libraries/apollo-gradle-plugin-external/src/main/kotlin/com/apollographql/apollo3/gradle/internal/DefaultApolloExtension.kt

```diff
@@ -662,16 +647,45 @@ abstract class DefaultApolloExtension(
       task.failOnWarnings.set(service.failOnWarnings)
 
       @Suppress("DEPRECATION")
-      val scalarTypeMappingFallbackOldSyntax = service.customScalarsMapping.orElse(
-          service.customTypeMapping
-      ).getOrElse(emptyMap())
-
-      check(service.scalarTypeMapping.isEmpty() || scalarTypeMappingFallbackOldSyntax.isEmpty()) {
-        "Apollo: either mapScalar() or customScalarsMapping can be used, but not both"
+      check(!service.customScalarsMapping.isPresent) {
+        """
+          Apollo: customScalarsMapping is deprecated. Use mapScalar() instead.
+          For an example, replace:
+          
+          customScalarsMapping = ["Date": "java.util.Date"]
+          or
+          customScalarsMapping.put("Date", "java.util.Date")
+          or
+          customScalarsMapping.set(mapOf("Date" to "java.util.Date"))
+          
+          With:
+          
+          mapScalar("Date", "java.util.Date")
+                    
+          If you have several scalars, call mapScalar() several times.
+        """.trimIndent()
```

Comment:
Behaviour change: using this will now trigger an error because from Groovy the deprecation warning/errors are not displayed